tests/unit/test_synapse_workload.py

```diff
@@ -253,6 +333,103 @@ def test_enable_federation_domain_whitelist_error(
         synapse.enable_federation_domain_whitelist(container_mock, harness.charm._charm_state)
 
 
+def test_enable_trusted_key_servers_no_action(harness: Harness):
+    """
+    arrange: set mock container with file.
+    act: call enable_trusted_key_servers without changing the configuration.
+    assert: configuration is not changed.
+    """
+    root = harness.get_filesystem_root(synapse.SYNAPSE_CONTAINER_NAME)
+    config_path = root / synapse.SYNAPSE_CONFIG_PATH[1:]
+    config_path.parent.mkdir(parents=True, exist_ok=True)
+    config_path.write_text(
+        """
+listeners:
+    - type: http
+      port: 8080
+      bind_addresses:
+        - "::"
+"""
+    )
+
+    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+    config = {"server_name": "foo"}
+    synapse_config = SynapseConfig(**config)  # type: ignore[arg-type]
+
+    synapse.enable_trusted_key_servers(
+        container, CharmState(datasource=None, saml_config=None, synapse_config=synapse_config)
+    )
+
+    with open(config_path, encoding="utf-8") as config_file:
+        content = yaml.safe_load(config_file)
+        expected_config_content = {
+            "listeners": [
+                {"type": "http", "port": 8080, "bind_addresses": ["::"]},
+            ],
+        }
+        assert content == expected_config_content
+
+
+def test_disable_room_list_search_success(harness: Harness):
+    """
+    arrange: set mock container with file.
+    act: change the configuration file.
+    assert: new configuration file is pushed and room_list_search is disabled.
+    """
+    root = harness.get_filesystem_root(synapse.SYNAPSE_CONTAINER_NAME)
+    config_path = root / synapse.SYNAPSE_CONFIG_PATH[1:]
+    config_path.parent.mkdir(parents=True, exist_ok=True)
+    config_path.write_text(
+        """
+listeners:
+    - type: http
+      port: 8080
+      bind_addresses:
+        - "::"
+"""
+    )
+
+    container: ops.Container = harness.model.unit.get_container(synapse.SYNAPSE_CONTAINER_NAME)
+
+    synapse.disable_room_list_search(container)
+
+    with open(config_path, encoding="utf-8") as config_file:
+        content = yaml.safe_load(config_file)
+        expected_config_content = {
+            "listeners": [
+                {"type": "http", "port": 8080, "bind_addresses": ["::"]},
+            ],
+            "enable_room_list_search": False,
+        }
+        assert content == expected_config_content
+
+
+def test_validate_config_error(monkeypatch: pytest.MonkeyPatch):
+    """
+    arrange: set mock container with invalid configuration file.
```

Comment:
Maybe rewrite the docstring to be more explicit, such as "mock the validation command to fail", as the test does not actually create an invalid configuration file.